Maskbook/Scene/App/RedPacket/ViewModel/LuckyDropViewModel.swift

```diff
@@ -0,0 +1,728 @@
+        }
+        let time = gasFeeItem.shortCostTime
+        
+        return "\(gasPrice) (\(time))"
+    }
+    
+    var gasLimit: BigUInt {
+        guard let gasLimitStr = gasFeeViewModel.gasFeePublisher.value?.gasLimit else {
+            return BigUInt(21_000)
+        }
+        return BigUInt(gasLimitStr) ?? BigUInt(21_000)
+    }
+    
+    var gasPrice: BigUInt? {
+        guard let gasFeeModel = gasFeeViewModel.gasFeePublisher.value else {
+            return nil
+        }
+        guard let gwei = BigUInt(gasFeeModel.gWei) else {
+            return nil
+        }
+        return gwei * (BigUInt(10).power(9))
+    }
+    
+    var totalQuantity: NSDecimalNumber {
+        var result: NSDecimalNumber
+        if mode == .average {
+            let quantity = NSDecimalNumber(string: quantityStr)
+            let amountPerShare = NSDecimalNumber(string: amountStr)
+            guard quantity != .notANumber && amountPerShare != .notANumber else {
+                return .zero
+            }
+            guard let decimal = token?.decimal else {
+                return .zero
+            }
+            result = quantity.multiplying(by: amountPerShare)
+            let roundBehavior = NSDecimalNumberHandler(
+                roundingMode: .down,
+                scale: decimal,
+                raiseOnExactness: false,
+                raiseOnOverflow: false,
+                raiseOnUnderflow: false,
+                raiseOnDivideByZero: false)
+            result = result.rounding(accordingToBehavior: roundBehavior)
+        } else {
+            result = NSDecimalNumber(string: amountStr)
+        }
+        return result != .notANumber ? result : .zero
+    }
+    
+    var totalQuantityStr: String {
+        let roundBehavior = NSDecimalNumberHandler(roundingMode: .plain, scale: 2, raiseOnExactness: false, raiseOnOverflow: false, raiseOnUnderflow: false, raiseOnDivideByZero: false)
+        let result = totalQuantity.rounding(accordingToBehavior: roundBehavior)
+        guard result != .notANumber else {
+            return "0.00"
+        }
+        let doubleValue = totalQuantity.doubleValue
+        if doubleValue < 0.01, doubleValue > 0 {
+            return "< 0.01"
+        }
+        return String(format: "%.2f", result.doubleValue)
+    }
+    
+    var totalQuantityColor: Color {
+        totalQuantity == .zero ?
+            Asset.Colors.Text.normal.asColor() :
+            Asset.Colors.Text.dark.asColor()
+    }
+    
+    var contractInfo: JSON? {
+        if let redPacketConstantURL = Bundle.main.url(forResource: "red-packet", withExtension: "json"),
+           let data = try? Data(contentsOf: redPacketConstantURL) {
+            return try? JSON(data: data)
+        } else {
+            return nil
+        }
+    }
+    
+    var luckyDropAddressStr: String? {
+        let chainKey = maskUserDefaults.network.redPacketConstantKey
+        return contractInfo?["HAPPY_RED_PACKET_ADDRESS_V4"][chainKey].string
```

Comment:
Use [this](https://github.com/DimensionDev/Mask-iOS/pull/63/files#diff-b77a5db49f20e019c27121cce6ed63ed0609d86bd1eaf5210b3f21d67346b7ecR28) instead.